src/formatDate.js

```diff
@@ -50,7 +50,55 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // happy checking
+
+  const inputArray = date.split('');
+  const result = [];
+
+  for (let i = 0; i < inputArray.length; i++) {
+    if (isNaN(inputArray[i] / 2)) {
+      inputArray[i] = ' ';
+    }
+  }
+
+  const threeElements = inputArray.join('').split(' ');
+
+  const YYwanted = toFormat.indexOf('YY') !== -1;
+  const YYYYwanted = toFormat.indexOf('YYYY') !== -1;
+
+  if (YYwanted) {
+    // change current format to match the desired
+    fromFormat[fromFormat.indexOf('YYYY')] = 'YY';
+
+    // change current date to match the desired
+    threeElements[fromFormat.indexOf('YY')]
+    = threeElements[fromFormat.indexOf('YY')].slice(2);
+  }
+
+  if (YYYYwanted) {
+    // change current format to match the desired
+    fromFormat[fromFormat.indexOf('YY')] = 'YYYY';
+
+    // change the current date to integer
+    threeElements[fromFormat.indexOf('YYYY')]
+    = parseInt(threeElements[fromFormat.indexOf('YYYY')]);
+
+    const currentDate = threeElements[fromFormat.indexOf('YYYY')];
+
+    if (currentDate < 30) {
+      threeElements[fromFormat.indexOf('YYYY')] += 2000;
+    } else if (currentDate < 100) {
+      threeElements[fromFormat.indexOf('YYYY')] += 1900;
```

Comment:
there is a lot of indexOf. each time that method is called it needs to traverse the entire array. accessing object values through keys is much more efficient. 

to me, a much simpler and faster solution would involve an object like 

{
  'DD': 13,
  'MM': 03,
  'YY': 2000,
}

Reply:
it's ontional tho

Reply:
Okay, I get your point. But changing to your proposed logic will require me to rethink this task all over again from scratch and for now, I would like to focus on learning other topics to not to lag behind